### PR TITLE
Asher/collab service

### PR DESCRIPTION
**Pull request by asherhy**
- [x] Build services for the editor function
  - [x] Add socket file
  - [x] Add MongoDB for persisting
  - [ ] Add auth for the socket
- [x] Add frontend files for the editor
- [x] Add frontend files for the collaboration page
- [x] Rename and reroute the page matching to match
- [x] Change functionality in match services such that user is redirected after 5 seconds after match to the collaboration page

### CODE REVIEW

File: editor-service/model/repository.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import EditorModel from './editor-model.js';
+import 'dotenv/config.js';
+
+// Set up mongoose connection
+import mongoose from 'mongoose';
+
+const mongoDB = process.env.ENV == 'PROD' ? process.env.DB_CLOUD_URI : process.env.DB_LOCAL_URI;
+
+mongoose.connect(mongoDB, { useNewUrlParser: true, useUnifiedTopology: true });
```
Comment (glennljw) [inline editor-service/model/repository.js:9]: I think for this we can specify `dbname` here to use different database?

File: editor-service/index.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+router.get('/', (_, res) => {
+  res.send('Hello World from collaboration-service');
+});
+
+app.use('/api/collaboration', router).all((_, res) => {
+  res.setHeader('content-type', 'application/json');
+  res.setHeader('Access-Control-Allow-Origin', '*');
+});
+
+const PORT = process.env.PORT || 8002;
```
Comment (glennljw) [inline editor-service/index.js:22]: Might want to change this to 8004 or something as user-service currently runs on 8000, matching-service on 8001, question-service on 8002 and history-service on 8003.

File: editor-service/model/repository.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    return editor;
+  }
+  return EditorModel.create({
+    _id: id,
+    username_first: 'a',
+    username_second: 'b',
+    session_id: 'aa',
+    data: defaultValue,
+  });
+}
```
Comment (glennljw) [inline editor-service/model/repository.js:29]: Can I check if this part is to integrate with matching later on?

**Review by glennljw — CHANGES_REQUESTED** [review]
LGTM so far! Thanks for the hard work, impressive job!

File: editor-service/model/repository.js (hunk 3 above)
Comment (asherhy) [reply editor-service/model/repository.js:29]: Yea, its for the auth part to check if user is authenticated to edit/view this editor

**Review by glennljw — APPROVED** [review]

**Pull request merged**